backend/dataall/modules/omics/db/omics_repository.py

```diff
@@ -0,0 +1,85 @@
+"""
+DAO layer that encapsulates the logic and interaction with the database for Omics
+Provides the API to retrieve / update / delete omics resources
+"""
+
+from sqlalchemy import or_
+from sqlalchemy.sql import and_
+from sqlalchemy.orm import Query
+
+from dataall.base.db import paginate, exceptions
+from dataall.modules.omics.db.omics_models import OmicsWorkflow, OmicsRun
+from dataall.core.environment.services.environment_resource_manager import EnvironmentResource
+
+
+class OmicsRepository:
```

Comment:
Should this inherit from `(EnvironmentResource)` similar to how we do for other dataall modules (i.e. notebooks, mlstudio, etc.)

Reply:
What we can then do is automatically start env stack update when Omics env feature is enabled for a particular environment

Currently one needs to manually trigger update stack after editing omics module for a particular env

Reply:
No, we don't need to block deletion of an Environment because of Omics runs. Runs do not have infra associated with them, so they can be deleted afterwards.

Reply:
I originally thought we were doing some type of dynamic permission additions when we enabled Omics environment feature - and thus it would be nice to automatically trigger a stack update by implementing `def update_env()` in `OmicsRepository(EnvironmentResource)`


I see that is not the case - so as of now I think it is fine to not inherit from `EnvironmentResource` but when moving to supporting private workflows and creating IaC in Omics from the console we may need to re-open that discussion

Resolving for now